src/IECoreScene/PrimitiveInterpolator.cpp

```diff
@@ -71,7 +73,49 @@ PrimitivePtr interpolatePrimitive( const Primitive *y0, const Primitive *y1, dou
 			it->second.interpolation == namedPrimitiveVariable.second.interpolation
 		)
 		{
-			ObjectPtr interpolatedData = linearObjectInterpolation( namedPrimitiveVariable.second.data.get(), it->second.data.get(), x );
+			const PrimitiveVariable &y0PrimVar = namedPrimitiveVariable.second;
+			const PrimitiveVariable &y1PrimVar = it->second;
+
+			const Data *y0Sample = y0PrimVar.data.get();
+			const Data *y1Sample = y1PrimVar.data.get();
+
+			size_t y0Size = IECore::size( y0Sample );
+			size_t y1Size = IECore::size( y1Sample );
+
+			if( y0Size != y1Size )
+			{
+				msg(
+					MessageHandler::Level::Error, "interpolatePrimitive", boost::str(
+						boost::format( "primitive variable '%s' data array size changes between samples. ( sample 0 size: %i sample 1 size: %i, alpha: %f )" ) %
+							it->first %
+							y0Size %
+							y1Size %
+							x
+					)
+				);
+				return nullptr;
```

Comment:
This means we're treating mismatched sizes and indices differently to mismatched types and interpolations. The latter two just result in a silently non-interpolated variable, whereas the former two now result in a message and a null result. Do we have a rationale for this, or would we be better to apply the same rules to both?

Reply:
In that case I think we should generate an error message in the mismatched size case. I can't see the type checking case you mention though.

Reply:
> I can't see the type checking case you mention though.

Line 69, right above this :

```
if( it != y1->variables.end() &&
    it->second.data->typeId() == namedPrimitiveVariable.second.data->typeId() &&
    it->second.interpolation == namedPrimitiveVariable.second.interpolation
)
```

If that fails, we just skip that variable and interpolate everything else we can. That seems to have been the general strategy prior to this PR (apart from the variableSize checks at line 49, which I think we want to be silent so that we load varying-topology data uninterpolated and then let a VelocityBlur node do the rest).